src/main/java/com/hotels/service/tracing/zipkintohaystack/forwarders/haystack/kinesis/KinesisForwarder.java

```diff
@@ -51,7 +56,12 @@ public void process(zipkin2.Span input) {
 
         byte[] value = span.toByteArray();
 
-        // TODO: metrics with success/failures
-        producer.putRecord(streamName, ByteBuffer.wrap(value), span.getTraceId());
+        try {
```

Comment:
AWS SDK 2 is nicer but it requires deeper changes. We should look into it at some point.